File: src/main/java/supercritical/api/capability/ICoolantHandler.java
package supercritical.api.capability;

import net.minecraft.core.Direction;
import net.minecraft.world.level.material.Fluid;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import supercritical.api.capability.impl.LockableFluidTank;
import supercritical.api.metatileentity.multiblock.IFissionReactorHatch;
import supercritical.api.nuclear.fission.ICoolantStats;

public interface ICoolantHandler extends ILockableHandler<Fluid>, IFissionReactorHatch {

    @Override
    boolean isLocked();

    @Override
    void setLock(boolean isLocked);

    @Nullable
    ICoolantStats getCoolant();

    void setCoolant(@Nullable ICoolantStats prop);

    @NotNull
    LockableFluidTank getFluidTank();

    @NotNull
    Direction getFrontFacing();

    @Nullable
    ICoolantHandler getOutputHandler();
}

File: src/main/kotlin/supercritical/api/capability/ICoolantHandler.kt
package supercritical.api.capability

import net.minecraft.core.Direction
import net.minecraft.world.level.material.Fluid
import supercritical.api.capability.impl.LockableFluidTank
import supercritical.api.machine.multiblock.IFissionReactorHatch
import supercritical.api.nuclear.fission.ICoolantStats

interface ICoolantHandler : ILockableHandler<Fluid?>, IFissionReactorHatch {

    var coolant: ICoolantStats?

    val fluidTank: LockableFluidTank

    val coolantFrontFacing: Direction

    val outputHandler: ICoolantHandler?
}
